.github/crystal.json

```
@@ -0,0 +1,29 @@
+{
+    "problemMatcher": [
+        {
+            "owner": "crystal",
+            "pattern": [
+                {
+                    "regexp": "^In (\\x1b\\[[0-9;]+m)?(.+):(\\d+):(\\d+)(\\x1b\\[[0-9;]+m)?$",
+                    "file": 2,
+                    "line": 3,
+                    "column": 4
+                },
+                {
+                    "regexp": "^$"
+                },
+                {
+                    "regexp": "^.+$"
+                },
+                {
+                    "regexp": "^.+$"
+                },
+                {
+                    "regexp": "^(\\x1b\\[[0-9;]+m)?(Error|Warning): (.+)(\\x1b\\[[0-9;]+m)?$",
```

In a regexp such as `(.+)(anything)?`, the `anything` part can never match as it will always go to the prior part. Probably need a non-greedy match, assuming that works 
```suggestion
                    "regexp": "^(\\x1b\\[[0-9;]+m)?(Error|Warning): (.+?)(\\x1b\\[[0-9;]+m)?$",
```

---

Good catch. It may be simpler to just drop that last group completely as the ANSI sequence will likely be scrubbed when the annotation is pushed. One sec - I'll give it a quick test.

---

Yep, seems to work! https://github.com/kimburgess/problem-matcher-test/actions/runs/954725001